pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -287,7 +287,7 @@ func validateImmutableFieldsCluster(new, old *Cluster) field.ErrorList {
 		}
 	}
 
-	if !old.IsSelfManaged() {
+	if !old.IsSelfManaged() || features.IsActive(features.ExperimentalSelfManagedClusterUpgrade()) {
 		clusterlog.Info("Cluster config is associated with workload cluster", "name", old.Name)
```

Comment:
this log doesn't make a lot of sense anymore
I would delete it
It doesn't add much info imo either